examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -91,27 +154,83 @@ void OTAProviderExample::SetOTAFilePath(const char * path)
     }
 }
 
+void OTAProviderExample::SetUserConsentDelegate(chip::ota::UserConsentDelegate * delegate)
+{
+    VerifyOrReturn(delegate != nullptr, ChipLogError(SoftwareUpdate, "User consent delegate is null"));
+    mUserConsentDelegate                      = delegate;
+    mUserConsentDelegate->userConsentCallback = OnUserConsent;
+}
+
 EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * commandObj,
                                                    const chip::app::ConcreteCommandPath & commandPath,
                                                    const QueryImage::DecodableType & commandData)
 {
-    // TODO: add confiuration for returning BUSY status
+    ChipLogDetail(SoftwareUpdate, "Requestor endpoint:%hu node-id:%llu", commandPath.mEndpointId,
+                  emberAfCurrentCommand()->SourceNodeId());
```

Comment:
Please don't use `emberAfCurrentCommand`.  If we need a SourceNodeId API on `CommandHandler`, please add one.

That said, what guarantees that this command was not groupcast (in which case SourceNodeId() can't provide a useful answer and the current impl would crash)?  It would be a weird thing to do, but we shouldn't crash because someone on the network did something weird.

Reply:
Resolving, as this stands outdated now.